src/main/java/net/ossindex/version/impl/AndRange.java

```diff
@@ -263,6 +263,15 @@ public String toString()
     return sb.toString();
   }
 
+  @Override
+  public String toMavenString()
+  {
+    StringBuilder sb = new StringBuilder();
+    sb.append(range1.toMavenString());
+    sb.append(range2.toMavenString());
+    return sb.toString().replace(",)(,",",");
```

Comment:
This code works for the common situation where range1 and range2 are "simple" ranges. It will not work for complex "and" cases. These complex cases are much less common and will not tend to be used in situations where the user will be interested in outputting a Maven type version range. I am not even really sure whether Maven (Set) range format can handle complex anded ranges.

For that reason we should restrict the use of this case to simple ranges only, and otherwise throw an exception.

`  @Override
  public String toMavenString()
  {
    if (range1.isSimple() && range2.isSimple()) {
      StringBuilder sb = new StringBuilder();
      sb.append(range1.toMavenString());
      sb.append(range2.toMavenString());
      return sb.toString().replace(",)(,", ",");
    }

    throw new UnsupportedOperationException();
  }
`